dapp/src/pages/AccessPage/loader.ts

```diff
@@ -0,0 +1,40 @@
+import { getAccessCodeFromLocalStorage } from "#/hooks/useAccessCode"
+import redirectWithSearchParams from "#/router/utils"
+import { acreApi, shouldDisplayWelcomeModal } from "#/utils"
+import { LoaderFunction } from "react-router-dom"
+
+export const redirectToAccessPageLoader = async (
+  url: string,
+  redirectToOnSuccess: null | string = null,
+  redirectToOnFail: null | string = null,
+) => {
+  try {
+    const isValid = await acreApi.verifyAccessCode(
+      getAccessCodeFromLocalStorage() ?? "",
+    )
```

Comment:
Even if `encodedCode` is not defined we send a request to `acreApi`.

```suggestion
    const encodedCode = getAccessCodeFromLocalStorage()
    const isValid = encodedCode
      ? await acreApi.verifyAccessCode(encodedCode)
      : false
```

Reply:
b98b7bf